queries.sql

```diff
@@ -1,3 +1,77 @@
 /*Queries that provide answers to the questions from all projects.*/
 
-SELECT * from animals WHERE name = 'Luna';
+SELECT * from animals WHERE name LIKE '%mon';
+SELECT name from animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
+SELECT name from animals WHERE neutered = TRUE AND escape_attempts <3;
+SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu';
+SELECT name, escape_attempts FROM animals WHERE weight_kg> 10.5;
+ SELECT * FROM animals WHERE neutered = TRUE;
+ SELECT * FROM animals WHERE name != 'Gabumon';
+ SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+  -- 2nd task
+  -- part 1
+BEGIN;
+-- try rollback
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+-- Update species column to pokemon for all animals without species.
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
+-- Verify that change was made and persists after commit.
+SELECT * FROM animals;
+
+-- part 3
+BEGIN;
+DELETE FROM animals;
+ROLLBACK
+SELECT * FROM animals;
+
+-- part 3
+BEGIN;
+DELETE FROM animals
+WHERE date_of_birth > 'Jan 1, 2022';
+
+SAVEPOINT deletion;
+
+UPDATE animals 
+SET weight_kg = weight_kg * -1;
+ROLLBACK TO deletion;
+
+UPDATE animals 
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+
+-- part 4 questions
+
+-- How many animals are there?
+SELECT COUNT(*) FROM animals;
+-- How many animals have never tried to escape?
+SELECT COUNT(escape_attempts) 
+FROM animals
+WHERE escape_attempts != 0;
+-- What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+-- Who escapes the most, neutered or not neutered animals?
+SELECT MAX(escape_attempts) FROM animals;
+-- What is the minimum and maximum weight of each type of animal?
+SELECT MAX(escape_attempts)
+FROM animals
+GROUP BY species;
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT AVG(escape_attempts)
+FROM animals
+WHERE date_of_birth BETWEEN 'Jan 1,1990' AND 'Dec 31,2000'
+GROUP BY species;
+
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
```

Comment:
- [ ] Good job is adding the screenshots, however you have not added those queries in this file. Make sure to add all 7 queries for:
```
What animals belong to Melody Pond?
List of all animals that are pokemon (their type is Pokemon).
List all owners and their animals, remember to include those that don't own any animal.
How many animals are there per species?
List all Digimon owned by Jennifer Orwell.
List all animals owned by Dean Winchester that haven't tried to escape.
Who owns the most animals?
```